src/storage/src/decode/mod.rs

```diff
@@ -158,8 +159,17 @@ impl PreDelimitedFormat {
         match self {
             PreDelimitedFormat::Bytes => Ok(Some(Row::pack(Some(Datum::Bytes(bytes))))),
             PreDelimitedFormat::Json => {
-                let j = mz_repr::adt::jsonb::Jsonb::from_slice(bytes)
-                    .map_err(|_| DecodeErrorKind::Text("Failed to decode JSON".to_string()))?;
+                let j = mz_repr::adt::jsonb::Jsonb::from_slice(bytes).map_err(|e| {
+                    DecodeErrorKind::Bytes(format!(
```

Comment:
Why do we want the additional `Bytes` variant? Only because we're decoding from bytes?

Reply:
Yes, I was just pattern matching on the existing structure

Reply:
yeah, not sure what the thinking behind those error kinds was, we technically already decoded from bytes before ... 😅

Reply:
Glad to make changes to this error in post if you have suggestions; this is gated behind unsafe/feature flags, so not a huge rush.